### ISSUE TITLE
[idea] [ios] Alert Controller

### ISSUE THREAD

**Issue by clsource** [liveviewnative/liveview-client-swiftui#34]
# Why?

AlertControllers (https://developer.apple.com/documentation/uikit/uialertcontroller?language=objc) are a way of presenting and requesting information to the user.

They are useful in Forms and other events.

## How?

This is just a sample syntax. Considering that this element only will be presented on a boolean condition only.
This syntax is inspired by https://surface-ui.org/

```xml
<alert :if={show_alert} title="My Title" message="My Message">
  <button style="cancel">Cancel</button>
  <button>OK</button>
</alert>
```

**Comment by bcardarella** [comment]
This appears to be part of UIKit and would be out of scope for the switui client.

I do think this is important to cover though and probably should be a part of a project that covers UIKit components that you can add as a dependency

**Comment by clsource** [comment]
It's also available in SwiftUI as a presentation modifier 
https://developer.apple.com/documentation/swiftui/view-presentation 

https://developer.apple.com/documentation/swiftui/view/alert(_:ispresented:actions:)-1bkka

**Comment by shadowfacts** [comment]
Alerts will definitely be needed, but the infrastructure isn't currently there to implement them. It's blocked by a couple things:

1. There needs to be a mechanism for turning DOM elements into modifiers that apply to the parent element
2. There needs to be a way of creating two-way bindings to assigns on the backend that are accessible from the client. Alerts in SwiftUI require a Binding to a boolean that controls whether the alert's presented or not, and it needs to be writable from the client (otherwise the alert can't be dismissed)

**Comment by stevegraham** [comment]
> There needs to be a mechanism for turning DOM elements into modifiers that apply to the parent element

What do you think of the same syntax as liveview slots, i.e. child element with prefixed tag name?

```
<view>
  // view content
  <:alert title="Attention!" is-presented="false">
    // alert content
  </:alert>
</view>
```

**Comment by AZholtkevych** [comment]
Duplicates #636 . Rejecting
